v2/googlecloud-to-neo4j/src/main/java/com/google/cloud/teleport/v2/neo4j/database/Neo4jConnection.java

```diff
@@ -124,28 +95,13 @@ public void resetDatabase() {
     // Direct connect utility...
     LOG.info("Resetting database");
     try {
-      String crdeCypher = "CREATE OR REPLACE DATABASE `neo4j`";
-      if (!StringUtils.isEmpty(database)) {
-        StringUtils.replace(crdeCypher, "neo4j", database);
-      }
-      LOG.info("Executing delete DB cypher: {}", crdeCypher);
-      executeCypher(crdeCypher);
-    } catch (Exception crde) {
-      LOG.error(
-          "Error executing reset database using CREATE OR REPLACE: {}, {}",
-          crde,
-          crde.getMessage());
-      // Trying DELETE /DETACH approach
-      try {
-        String ddeCypher = "MATCH (n) DETACH DELETE n";
-        LOG.info("Executing alternate delete cypher: {}", ddeCypher);
-        executeCypher(ddeCypher);
-        String constraintsDeleteCypher =
-            "CALL apoc.schema.assert({},{},true) YIELD label, key RETURN *";
-        executeCypher(constraintsDeleteCypher);
-      } catch (Exception dde) {
-        LOG.error("Error executing detach delete", dde);
-      }
+      String database = !StringUtils.isEmpty(this.database) ? this.database : "neo4j";
+      String cypher = String.format("CREATE OR REPLACE DATABASE `%s`", database);
```

Comment:
Isn't this subject to injection?

Reply:
Yes, good catch. Escaping would be good to prevent issues.

The attack vector might be nonexistent though -- those parameters come from the given template options, and in theory for you to inject something / actually run the statement, the credentials (that are part of the same payload) have to work -- so they already have privileged access / no need for escalation through injection.

Reply:
note: it's as vulnerable as it was before the change, but let me switch to the safer parameterized query instead.

Reply:
thanks for the catch @jonathan-lemos